### PR TITLE
Add support for BigInt

### PR DESCRIPTION
**Pull request by kevcenteno**
This adds support for `BigInt` which is in Node v10.4+

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@ node_modules
 coverage
 npm-debug.log
 .nyc_output
+yarn.lock
```
Comment (runk) [inline .gitignore]: Why? This module doesn't use yarn :/

File: lib/decoder.js [hunk 2]
```diff
@@ -281,12 +281,24 @@ Decoder.prototype.decodeUint = function(offset, size) {
 
 Decoder.prototype.decodeBigUint = function(offset, size) {
   var buffer = new Buffer(size);
+  var hasBigInt = typeof BigInt === 'function';
```
Comment (runk) [inline lib/decoder.js]: I think it should be done once, and configured as a helper fn at `Decoder` initialization

Something like

```
this.bigIntHelper = typeof BigInt === 'function' ?
   nativeBigIntHandler : 
   thirdPartyBigIntHandler
```

And then just use it where needed - it'll also help with testing.

Comment (kevcenteno) [reply lib/decoder.js]: Good call.  I didn't end up writing a helper function because the arithmetic APIs differ, so for now it's just a boolean.

**Comment by runk** [comment]
Alright, merging this one into feature branch, will fix build myself.

**Comment by runk** [comment]
Thanks for the contribution!

**Pull request merged**